tools/parsergen/src/parsergen/bsl_codegen.py

```
@@ -0,0 +1,707 @@
+            body = self._render_alternative(alternative, "\t\t")
+            if body:
+                result.extend(body.rstrip("\r\n").split("\r\n"))
+
+        has_epsilon = any(
+            not alternative.syntax_symbols
+            for alternative in production.alternatives
+        )
+        if dispatchable and not has_epsilon:
+            result.extend(
+                (
+                    "\tИначе",
```

Ruff (0.16.1)

[warning] String contains ambiguous `а` (CYRILLIC SMALL LETTER A). Did you mean `a` (LATIN SMALL LETTER A)?

(RUF001)